client/client.go

```diff
@@ -150,6 +152,37 @@ func FromEnv(c *Client) error {
 	return nil
 }
 
+// WithContextStoreOrEnv applies a context store config, or classic docker env if detected
+// Heuristic to use either context store or legacy DOCKER_* variables is:
+// if contextNameOverride is set -> load this context from context store
+// if DOCKER_CONTEXT env variable is set -> load this context from context store
+// if DOCKER_HOST is set -> fallback to FromEnv
+// if a current context is set in the store -> load this context
+// fallback to "default" client
+func WithContextStoreOrEnv(contextStoreDir, contextNameOverride string) func(*Client) error {
```

Comment:
I appreciate there is precedent here in `WithTLSClientConfig` but it seems a bit odd to me to be requiring all callers to opt in to what really ought to be the default behaviour.

IOW -- perhaps the functionality here ought to be folded in to `NewClientWithOpts` (in place of the call to `defaultHTTPClient`).

More a question for the maintainers rather than the author I guess.

Reply:
Yeah... the thing is it also introduce the 2 new parameters. Do we really want every client code to specify those, even if at the end they do things like `WithHost("/var/run/docker.sock")` ?

Reply:
They both have sensible defaults, so could be separate independent `WithContextStore` and `WithContextName`, but without them the default could be to use the default/current context from the default store.

So (IMHO) `NewClientWithOptions()` with no `WithFoo` options given should DTRT.

Reply:
I am waiting for feedback on this. I know that at the time being, for exemple, NewClientWithOptions without any parameter does not look for env variables. I am not sure if we want to change that.